service/src/main/java/gov/nasa/pds/api/registry/search/RegistrySearchRequestBuilder.java

```diff
@@ -53,222 +52,187 @@ public class RegistrySearchRequestBuilder {
-
-    MatchQuery lidvidMatch = new MatchQuery.Builder().field("_id").query(lidvidFieldValue).build();
-
-    this.must.add(lidvidMatch.toQuery());
-
-    return this;
-
+  /**
+   * Add a constraint that a given field name must match the given field value
+   * @param fieldName the name of the field in OpenSearch format
+   * @param identifier the PDS identifier whose string representation must be present in the given field
+   */
+  public RegistrySearchRequestBuilder matchField(String fieldName, PdsProductIdentifier identifier) {
+    return this.matchField(fieldName, identifier.toString());
   }
 
-
-  public RegistrySearchRequestBuilder addLidMatch(PdsProductIdentifier identifier) {
-    // lid match
-    FieldValue lidvidFieldValue =
-        new FieldValue.Builder().stringValue(identifier.getLid().toString()).build();
-
-    MatchQuery lidMatch = new MatchQuery.Builder().field("lid").query(lidvidFieldValue).build();
-
-    this.must.add(lidMatch.toQuery());
-
-    return this;
-
+  public RegistrySearchRequestBuilder matchLidvid(PdsProductIdentifier identifier) {
+    return this.matchField("_id", identifier);
   }
 
-  public RegistrySearchRequestBuilder paginates(Integer limit, List<String> sort,
-      List<String> searchAfter) throws SortSearchAfterMismatchException {
+  public RegistrySearchRequestBuilder matchLid(PdsProductIdentifier identifier) {
+    return this.matchField("lid", identifier);
+  }
 
-    if ((sort != null) && (!sort.isEmpty())) {
-      this.sort(sort);
+  public RegistrySearchRequestBuilder paginate(Integer pageSize, List<String> sortFieldNames,
+      List<String> searchAfterFieldValues) throws SortSearchAfterMismatchException {
+    if ((sortFieldNames != null) && (!sortFieldNames.isEmpty())) {
+      this.sortFromStrings(sortFieldNames);
     }
 
-    this.size(limit);
+    this.size(pageSize);
 
-    if ((searchAfter != null) && (!searchAfter.isEmpty())) {
-      if (sort == null) {
+    if ((searchAfterFieldValues != null) && (!searchAfterFieldValues.isEmpty())) {
+      if (sortFieldNames == null) {
         throw new SortSearchAfterMismatchException("sort argument must be provided if searchAfter argument is provided");
-      } else if (searchAfter.size() != sort.size()) {
+      } else if (searchAfterFieldValues.size() != sortFieldNames.size()) {
         throw new SortSearchAfterMismatchException("sort and searchAfter arguments must be of equal length if provided");
       }
-      this.searchAfter(searchAfter);
+      this.searchAfterFromStrings(searchAfterFieldValues);
     }
 
     return this;
 
   }
 
-  public RegistrySearchRequestBuilder sort(List<String> sort) {
+  /**
+   * Implements an alternative to .sort() that accepts strings in API property format.
+   * Currently hardcoded to sort in ascending order only.
+   * @param sortFieldNames
+   */
+  public RegistrySearchRequestBuilder sortFromStrings(List<String> sortFieldNames) {
```

Comment:
Do we need the fromString in the method name ? It is in the prototype of the function.

Reply:
Now I understand, it conflicts with the sort of the super class.

Reply:
But we might keep the same name for both, did you try ? the polymorphism of java should solve that...

Reply:
[Necessary, unfortunately](https://jpl.slack.com/archives/C023TQGPW4A/p1720720188360579)

I'd have preferred a polymorphic interface too but ya gets what ya gets

Reply:
That's is too complicated for me to follow the details, but I can feel that there is some background here. Thanks !

Reply:
The short version is "Java doesn't support polymorphism where the only difference is in the type `T` of some parameter generic  `SomeType<T>`

So `List<String>` and `List<PdsProductIdentifier>` are the same type from Java's perspective, when checking for conflicts.